server/src/models/tasks.ts

```diff
@@ -0,0 +1,89 @@
+import knex from "../pool";
+
+export async function createTask(newTask: any) {
+    return await knex('tasks')
+        .insert(newTask)
+        .returning('id')
+        .then(([{id}]) => {
+            newTask['id'] = id;
+            return newTask;
+        })
+        .catch((err) => {
+            throw new Error('Failed to create new Task ' + err);
+        })
+}
+
+// I have no idea if any of these work
+export async function updateTask(newTask: any) {
+  return await knex("tasks")
+    .where("id", newTask.id)
+    // chain updates?
+    .update("info_type", newTask.info_type)
+    .update("info_id", newTask.info_id)
+    .update("assigner_id", newTask.assigner_id)
+    .update("assignee_id", newTask.assignee_id)
+    .update("due_date", newTask.due_date)
+    .update("created_date", newTask.created_date)
+    .returning("id")
+    .then(([{id}]) => { // not sure if this works, just copied from createTask (thanks Rohit)
+      return newTask;
+  })
+    .catch((err) => {
+      throw new Error("Failed to update existing Task " + err);
+    })
+}
```

Comment:
```suggestion
export async function updateTask(attributes: any) {
  const { id, info: info_attributes, ...attributes } = attributes; // extract id and k-specific fields
  const task = await knex('tasks').where({ id }).update(attributes, "*"); // update the task
  task.info = await knex(task.info_type).where({ id: task.info_id }).update(info_attributes, "*"); // update the associated info
  return task;
}
```

This is the most basic example of what you want to implement here. You should assume that the request body will be a task with the addition of an entry for `info`, which will contain the fields for the associated info. Knowing them all isn't necessary here, you can just forward them to the database.

Reply:
Also, `await` and `then` are two ways of doing the same thing (asyncronous JS), so you should be using one or the other.